File: hello-world-samples/scala01/src/main/scala/com/mvv/bank/orders/rest/conversion/impl/CashOrderMapper.scala
package com.mvv.bank.orders.rest.conversion.impl

import com.mvv.bank.orders.domain.CashOrder as DomainCashOrder
import com.mvv.bank.orders.rest.entities.CashOrder as DtoCashOrder
import org.mapstruct.{Mapper, Mapping}


@Mapper(uses = Array(classOf[CurrencyMapper])) // now this annotation is ignored and similar one from java class is used
trait CashOrderMapper {
  @Mapping(source = "domainField", target = "dtoField")
  def toDto(source: DomainCashOrder): DtoCashOrder

  @Mapping(source = "dtoField", target = "domainField")
  def toDomain(source: DtoCashOrder): DomainCashOrder
}

File: hello-world-samples/scala01/src/main/scala/com/mvv/bank/orders/rest/conversion/CashOrderMapper.java
package com.mvv.bank.orders.rest.conversion;

import org.mapstruct.Mapper;


@Mapper(uses = {CurrencyMapper.class, AmountMapper.class, OptionMapper.class})
public interface CashOrderMapper
        extends com.mvv.bank.orders.rest.conversion.impl.CashOrderMapper {
}
